primitives/src/select/components/select.rs

```diff
@@ -0,0 +1,164 @@
+//! Main Select component implementation.
+
+use crate::{use_controlled, use_effect};
+use dioxus::prelude::*;
+use dioxus_core::Task;
+
+use super::super::context::{SelectContext, SelectValue};
+use crate::focus::use_focus_provider;
+
+/// Props for the main Select component
+#[derive(Props, Clone, PartialEq)]
+pub struct SelectProps<T: Clone + PartialEq + 'static> {
+    /// The controlled value of the select
+    #[props(default)]
+    pub value: ReadOnlySignal<Option<Option<SelectValue<T>>>>,
+
+    /// The default value of the select
+    #[props(default)]
+    pub default_value: Option<SelectValue<T>>,
+
+    /// Callback when the value changes
+    #[props(default)]
+    pub on_value_change: Callback<Option<SelectValue<T>>>,
+
+    /// Whether the select is disabled
+    #[props(default)]
+    pub disabled: ReadOnlySignal<bool>,
+
+    /// Whether the select is required
+    #[props(default)]
+    pub required: ReadOnlySignal<bool>,
```

Comment:
During refactoring I mention that this property is unused. Should it be this way?

Also I mentioned that we using simple `div` instead of HTML `select`. Could you explain me why?

Reply:
I removed the prop. It would be good to support in the future, but it shouldn't exist while it does nothing.

We don't use the built in select elements because it doesn't let you style the dropdown